eru/src/store/modules/authentication/actions.js

```diff
@@ -137,6 +137,27 @@ export default {
     // console.log("data from the actions module of auth=>", data);
   },
 
+  //action to get the current user data
```

Comment:
Yes it is an action i created to be able to get the current user's data from the backend. so that we that need rewrite the same code to get the user from the backend in several places

Reply:
> Yes it is an action i created to be able to get the current user's data from the backend. so that we that need rewrite the same code to get the user from the backend in several places

There was not existing action for that in the store?

Reply:
> > Yes it is an action i created to be able to get the current user's data from the backend. so that we that need rewrite the same code to get the user from the backend in several places
> 
> There was not existing action for that in the store?
Yes there was no action for that

Reply:
The only actions which were there were actions to login using different providers

Reply:
> > > Yes it is an action i created to be able to get the current user's data from the backend. so that we that need rewrite the same code to get the user from the backend in several places
> > 
> > 
> > There was not existing action for that in the store?
> > Yes there was no action for that

Have you not seen the getSpecificUserAction mothod in user module in the store?

Reply:
> > > > Yes it is an action i created to be able to get the current user's data from the backend. so that we that need rewrite the same code to get the user from the backend in several places
> > > 
> > > 
> > > There was not existing action for that in the store?
> > > Yes there was no action for that
> 
> Have you not seen the getSpecificUserAction mothod in user module in the store?

Yes i saw it, but it does not set the currentUserObject in the state

Reply:
> > > > > Yes it is an action i created to be able to get the current user's data from the backend. so that we that need rewrite the same code to get the user from the backend in several places
> > > > 
> > > > 
> > > > There was not existing action for that in the store?
> > > > Yes there was no action for that
> > 
> > 
> > Have you not seen the getSpecificUserAction mothod in user module in the store?
> 
> Yes i saw it, but it does not set the currentUserObject in the state

please, why not just do it (set the currentUserObject in the state) in place of creating another method?

Reply:
> > > > > > Yes it is an action i created to be able to get the current user's data from the backend. so that we that need rewrite the same code to get the user from the backend in several places
> > > > > 
> > > > > 
> > > > > There was not existing action for that in the store?
> > > > > Yes there was no action for that
> > > 
> > > 
> > > Have you not seen the getSpecificUserAction mothod in user module in the store?
> > 
> > 
> > Yes i saw it, but it does not set the currentUserObject in the state
> 
> please, why not just do it (set the currentUserObject in the state) in place of creating another method?

ok let me try that

Reply:
But i will have to do some modifications to the getSpecificUserAction to be able to use it in the router/index.js file. I can't access the Vue _vm from there

Reply:
I way you could go ahead this is to pass the _vm down as prop go the function link i did in some functions in the store

Reply:
> But i will have to do some modifications to the getSpecificUserAction to be able to use it in the router/index.js file. I can't access the Vue _vm from there

why can you not access  the Vue _vm? I am seeing something like payload._vm inside the method

Reply:
> I way you could go around this is to pass the _vm down as prop go the function link i did in some functions in the store

Yes, what I am saying is that the getSpecificUserAction requires the _vm as an argument, but i need to call that action from the router.js file and there is no way for me to access the current _vm from the router/index.js file.

Reply:
> > I way you could go around this is to pass the _vm down as prop go the function link i did in some functions in the store
> 
> Yes, what I am saying is that the getSpecificUserAction requires the _vm as an argument, but i need to call that action from the router.js file and there is no way for me to access the current _vm from the router/index.js file.

ok nice, your new method has a purpose

Reply:
Why does he need to call the function in the router again? Don't really get that point though

Reply:
> > > > > > Yes it is an action i created to be able to get the current user's data from the backend. so that we that need rewrite the same code to get the user from the backend in several places
> > > > > 
> > > > > 
> > > > > There was not existing action for that in the store?
> > > > > Yes there was no action for that
> > > 
> > > 
> > > Have you not seen the getSpecificUserAction mothod in user module in the store?
> > 
> > 
> > Yes i saw it, but it does not set the currentUserObject in the state
> 
> please, why not just do it (set the currentUserObject in the state) in place of creating another method?

And another advantage of the action i created is that we can get the current user using just the token.
- I am facing difficulties using the getSpecificUserAction, because it requires the _vm and the username, but I don't have access to all of that in the router/index.js file

Reply:
> > > > > > > Yes it is an action i created to be able to get the current user's data from the backend. so that we that need rewrite the same code to get the user from the backend in several places
> > > > > > 
> > > > > > 
> > > > > > There was not existing action for that in the store?
> > > > > > Yes there was no action for that
> > > > 
> > > > 
> > > > Have you not seen the getSpecificUserAction mothod in user module in the store?
> > > 
> > > 
> > > Yes i saw it, but it does not set the currentUserObject in the state
> > 
> > 
> > please, why not just do it (set the currentUserObject in the state) in place of creating another method?
> 
> And another advantage of the action i created is that we can get the current user using just the token.
> 
> * I am facing difficulties using the getSpecificUserAction, because it requires the _vm and the username, but I don't have access to all of that in the router/index.js file

Ok, I think it's ok, I'll approve this PR
So let me test it before

Reply:
> Why does he need to call the function in the router again? Don't really get that point though

So that we get the most recent value for the user object each time the site is reloaded

Reply:
Oh yeah, @viryl15 i think he's right 👍🏾

Reply:
I haven't reviewed his code in detail...  But i believe he makes a point... @viryl15 i trust you'll review it good and approve once done... I'm not home at the moment

Reply:
I tested and it works